kv/index.go

```diff
@@ -549,15 +549,21 @@ func (i *Index) verify(ctx context.Context, store Store, includeMissingSource bo
 // indexWalk consumes the indexKey and primaryKey pairs in the index bucket and looks up their
 // associated primaryKey's value in the provided source bucket.
 // When an item is located in the source, the provided visit function is called with primary key and associated value.
-func indexWalk(indexCursor ForwardCursor, sourceBucket Bucket, visit VisitFunc) (err error) {
+func indexWalk(ctx context.Context, indexCursor ForwardCursor, sourceBucket Bucket, visit VisitFunc) (err error) {
 	defer func() {
 		if cerr := indexCursor.Close(); cerr != nil && err == nil {
 			err = cerr
 		}
 	}()
 
 	for ik, pk := indexCursor.Next(); ik != nil; ik, pk = indexCursor.Next() {
-		v, err := sourceBucket.Get(pk)
+		// TODO(george): this is a work-around as lots of calls to Get()
+		// on a transaction causes issues with a particular implementation
+		// of kv.Store.
+		// The use of a cursor on this store bypasses the transaction
+		// and gives us the access pattern we desire.
+		// Please do not change back to a bucket.Get().
```

Comment:
yeah, this also changes the consistency guarantees, but that's ok, we don't want strong consistency